tests/spelling.R

```diff
@@ -0,0 +1,3 @@
+if (requireNamespace("spelling", quietly = TRUE)) {
```

Comment:
This seems like it belongs in a precommit check, rather than a formal package check. We should discuss this further.

Reply:
Before adding this, I chatted with @cicdguy, and I thought it could have been useful, also for local checks. We can discuss this today! We can also take it out, as @cicdguy already has in place a nice automatic spelling check

Reply:
after our discussion, removing the file and dep